packages/camera/camera_android/android/src/main/java/io/flutter/plugins/camera/media/MediaRecorderBuilder.java

```diff
@@ -75,33 +75,48 @@ public MediaRecorder build() throws IOException, NullPointerException, IndexOutO
     if (enableAudio) mediaRecorder.setAudioSource(MediaRecorder.AudioSource.MIC);
     mediaRecorder.setVideoSource(MediaRecorder.VideoSource.SURFACE);
 
-    if (Build.VERSION.SDK_INT >= 31) {
+    boolean isFormatNeedSetup = true;
+    boolean isAudioNeedSetup = enableAudio;
+    boolean isVideoNeedSetup = true;
+
+    if (Build.VERSION.SDK_INT >= 31 && encoderProfiles != null) {
```

Comment:
The changes here seem to suggest that there may be a case where one of `videoProfile` or `audioProfile` may be null but not the other. This does not seem to be true according to [the docs](https://developer.android.com/reference/android/media/EncoderProfiles).

Thus, you could add a check here for `if (Build.VERSION.SDK_INT >= 31 && encoderProfiles != null)` to use the non-deprecated code versus just `if (Build.VERSION.SDK_INT >= 31)` and alleviate the need for the `isXsetup` type of checks.

In order for that to work (and I think the code you have here, as well, since one of `encoderProfiles` and `camcorderProfile` will always be null), we'll need to handle using the correct `MediaRecorderBuilder` constructor at the `Camera.java` level:
https://github.com/flutter/plugins/blob/0a0e3d205ca32fdeebe237262e6ac824b3f3d51f/packages/camera/camera_android/android/src/main/java/io/flutter/plugins/camera/Camera.java#L261-L265

Reply:
I left a comment above in `ResolutionFeature.java` that may help determine which constructor to use here.